Sources/App/Models/Version.swift

```diff
@@ -113,7 +113,13 @@ final class Version: Model, Content {
 
 extension Version: Equatable {
     static func == (lhs: Version, rhs: Version) -> Bool {
-        lhs.id == rhs.id
+        if let id1 = lhs.id, let id2 = rhs.id {
+            return id1 == id2
+        } else {
+            return lhs.commit == rhs.commit
+                && lhs.commitDate == rhs.commitDate
+                && lhs.reference == rhs.reference
```

Comment:
This was a nasty issue for our test assertions: there were several cases where we simply created versions like so without saving:

```
let v = Version(package: p)
```

i.e. without `save()` and without assigning `id:` which meant we were comparing different versions with `nil` ids...

Reply:
(I don't think it was any existing cases but new ones I just wrote for this PR)